crates/forge/src/test_filter.rs

```diff
@@ -2,11 +2,13 @@ use crate::collecting::{CompiledTestCrate, CompiledTestCrateRaw, ValidatedForkCo
 use crate::TestCaseFilter;
 use test_collector::TestCase;
 
+use crate::shared_cache::get_cached_failed_tests_names;
+
 #[derive(Debug, PartialEq)]
 // Specifies what tests should be included
 pub struct TestsFilter {
-    // based on name
-    name_filter: NameFilter,
+    // based on names
+    name_filters: Vec<NameFilter>,
```

Comment:
Cannot we just have a separate bool/enum with two variansts like `last_failed`? Much less confusing imo